pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -280,9 +286,36 @@ func (e *cloudWatchExecutor) newSession(ctx context.Context, pluginCtx backend.P
 	// work around until https://github.com/grafana/grafana/issues/39089 is implemented
 	if e.cfg.SecureSocksDSProxy.Enabled && instance.Settings.SecureSocksProxyEnabled {
 		// only update the transport to try to avoid the issue mentioned here https://github.com/grafana/grafana/issues/46365
-		sess.Config.HTTPClient.Transport = instance.HTTPClient.Transport
-	}
+		var tr *http.Transport
+		var ok bool
+		if sess.Config.HTTPClient.Transport != nil {
+			tr, ok = sess.Config.HTTPClient.Transport.(*http.Transport)
+			if !ok {
+				return nil, errors.New("session http client transport is not of type http.Transport")
+			}
+		} else {
+			// following go standar library logic (https://pkg.go.dev/net/http#Client), if no Transport is provided,
+			// then we use http.DefaultTransport
+			trTmp, ok := http.DefaultTransport.(*http.Transport)
+			if !ok {
+				//this should not happen but validating just in case
+				return nil, errors.New("default http client transport is not of type http.Transport")
+			}
+			tr = trTmp.Clone()
+		}
+
+		dialer, err := proxy.New(instance.ProxyOpts).NewSecureSocksProxyContextDialer()
+		if err != nil {
+			return nil, fmt.Errorf("error building Secure socks proxy dialer: %w", err)
+		}
+		contextDialer, ok := dialer.(goproxy.ContextDialer)
+		if !ok {
+			return nil, err
+		}
```

Comment:
nit: I think we could use `proxy.New(...).ConfigureSecureSocksHTTPProxy(tr)` here, which would be a bit tidier and avoid the `goproxy` import

Reply:
you are right, it ended up being basically the same code :O 
fixed in ba59f4a